src/common/mod.rs

```diff
@@ -323,6 +323,43 @@ where
         Poll::Ready(Ok(pos))
     }
 
+    fn poll_write_vectored(mut self: Pin<&mut Self>, cx: &mut Context<'_>, bufs: &[IoSlice<'_>]) -> Poll<Result<usize, Error>> {
+        let mut pos = 0;
+        let total_length = bufs.iter().map(|s| s.len()).sum();
+
+        while pos != total_length {
+            let mut would_block = false;
+
+            match self.session.writer().write_vectored(&bufs) {
```

Comment:
we need to [advance it](https://doc.rust-lang.org/stable/std/io/struct.IoSlice.html#method.advance), or just call `write_vectored` once.

Reply:
Sounds like we are blocked on `advance_slices` actually being stabilised here :(

Reply:
We can implement it ourselves, or stop looping, just call `write_vectored` once.

Reply:
We can't advance it because we only have an immutable reference to the slice of buffers.